app/main.py

```diff
@@ -2,5 +1,17 @@
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args: list) -> Any:
+        if args in storage:
+            print("Getting from cache")
+            return storage[args]
+        else:
+            result = func(*args)
+            storage[args] = result
```

Comment:
Storing `args` as a key in the dictionary has the same issue as mentioned before. Ensure that `args` is hashable and does not contain any mutable elements.